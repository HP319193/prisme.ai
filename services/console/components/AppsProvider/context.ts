import { createContext, useContext } from 'react';

export interface AppsContext {
  apps: Map<string, Prismeai.App>;
  appInstances: Map<string, Prismeai.DetailedAppInstance[]>;
  getApps: ({
    text,
    page,
    limit,
    workspaceId,
  }: {
    text?: PrismeaiAPI.SearchApps.QueryParameters['text'];
    page?: PrismeaiAPI.SearchApps.QueryParameters['page'];
    limit?: PrismeaiAPI.SearchApps.QueryParameters['limit'];
    workspaceId?: PrismeaiAPI.SearchApps.QueryParameters['workspaceId'];
  }) => Promise<Prismeai.App[] | null>;
  getAppInstances: (
    workspaceId: PrismeaiAPI.ListAppInstances.PathParameters['workspaceId']
  ) => Promise<Prismeai.DetailedAppInstance[]>;
}

export const appsContext = createContext<AppsContext>({
  apps: new Map(),
  appInstances: new Map(),
  getApps: async () => ({} as any),
  getAppInstances: async () => ({} as any),
});

export const useApps = () => useContext(appsContext);

export default appsContext;
