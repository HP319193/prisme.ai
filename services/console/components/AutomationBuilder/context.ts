import { Schema } from '@prisme.ai/design-system';
import { createContext, useContext } from 'react';

export interface AutomationBuilderContext {
  automationId: string;
  addInstruction: (parent: Prismeai.InstructionList, index: number) => void;
  removeInstruction: (parent: Prismeai.InstructionList, index: number) => void;
  editInstruction: (parent: Prismeai.InstructionList, index: number) => void;
  editCondition: (
    parent: { conditions: Prismeai.Conditions },
    key: string
  ) => void;
  removeCondition: (
    parent: { conditions: Prismeai.Conditions },
    key: string
  ) => void;
  editTrigger: () => void;
  editOutput: () => void;
  getApp: (
    instruction: string
  ) => {
    name: string;
    icon: string;
    instructionName: string;
  };
  instructionsSchemas: [
    string,
    Record<string, Schema & { description?: string }>,
    { icon: string }
  ][];
  getSchema: (name: string) => Schema;
}
export const automationBuilderContext = createContext<AutomationBuilderContext>(
  {
    automationId: '',
    addInstruction() {},
    removeInstruction() {},
    editInstruction() {},
    editCondition() {},
    removeCondition() {},
    editTrigger() {},
    editOutput() {},
    getApp() {
      return {
        name: '',
        icon: '',
        instructionName: '',
      };
    },
    instructionsSchemas: [],
    getSchema() {
      return {} as Schema;
    },
  }
);

export const useAutomationBuilder = () => useContext(automationBuilderContext);

export default automationBuilderContext;
