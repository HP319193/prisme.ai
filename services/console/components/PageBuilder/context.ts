import { createContext } from 'react';
import { Schema } from '@prisme.ai/design-system';
import { BlockInCatalog } from './useBlocks';
import { useContext } from '../../utils/useContext';

export type Block = NonNullable<Prismeai.Page['blocks']>[number];
export type BlocksWithKeys = Map<string, Block>;

export interface PageBuilderContext {
  value: BlocksWithKeys;
  addBlock: (position: number, blockName?: string) => void;
  setEditBlock: (blockId: string) => void;
  removeBlock: (key: string) => void;
  setBlockConfig: (key: string, config: any) => void;
  setBlockSchema: (blockId: string, schema: Schema | null) => void;
  catalog: BlockInCatalog[];
  blocksSchemas: Map<string, Schema | null>;
}

export const context = createContext<PageBuilderContext | undefined>(undefined);

export const usePageBuilder = () => useContext<PageBuilderContext>(context);
