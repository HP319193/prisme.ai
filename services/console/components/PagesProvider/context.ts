import { createContext, Dispatch, SetStateAction, useContext } from 'react';

type WorkspaceId = string;

export interface PagesContext {
  pages: Map<WorkspaceId, Set<Prismeai.Page>>;
  fetchPages: (workspaceId: NonNullable<Prismeai.Workspace['id']>) => void;
  setPages: Dispatch<SetStateAction<Map<string, Set<Prismeai.Page>>>>;
  createPage: (
    workspaceId: NonNullable<Prismeai.Workspace['id']>,
    p: Prismeai.Page
  ) => Promise<Prismeai.Page>;
  savePage: (
    workspaceId: NonNullable<Prismeai.Workspace['id']>,
    p: Prismeai.Page
  ) => Promise<Prismeai.Page>;
  deletePage: (
    workspaceId: NonNullable<Prismeai.Workspace['id']>,
    pid: NonNullable<Prismeai.Page['id']>
  ) => Promise<Pick<Prismeai.Page, 'id'>>;
}

export const pagesContext = createContext<PagesContext>({
  pages: new Map(),
  setPages() {},
  fetchPages() {},
  async createPage() {
    return {} as Prismeai.Page;
  },
  async savePage() {
    return {} as Prismeai.Page;
  },
  async deletePage() {
    return {} as Prismeai.Page;
  },
});

export const usePages = () => useContext(pagesContext);

export default usePages;
