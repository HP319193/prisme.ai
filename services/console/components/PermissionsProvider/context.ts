import { createContext, useContext } from 'react';
import UserPermissions = Prismeai.UserPermissions;
import SubjectType = PrismeaiAPI.GetPermissions.Parameters.SubjectType;

export interface PermissionsContext {
  usersPermissions: Map<string, UserPermissions[]>;
  addUserPermissions: (
    subjectType: SubjectType,
    subjectId: string,
    permissions: UserPermissions
  ) => Promise<any>;
  // unshareWorkspaceToUser: (
  //   userEmail: string,
  //   workspaceId: string
  // ) => Promise<any>;
  getUsersPermissions: (
    subjectType: SubjectType,
    subjectId: string
  ) => Promise<UserPermissions[]>;
}

export const workspacesContext = createContext<PermissionsContext>({
  usersPermissions: new Map(),
  // unshareWorkspaceToUser: async () => ({} as any),
  addUserPermissions: async () => ({} as UserPermissions),
  getUsersPermissions: async () => [] as UserPermissions[],
});

export const usePermissions = () => useContext(workspacesContext);

export default workspacesContext;
