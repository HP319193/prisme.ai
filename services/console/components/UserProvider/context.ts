import { createContext, useContext } from 'react';
import { ApiError } from '@prisme.ai/sdk';

export enum OperationSuccess {
  emailSent = 'emailSent',
  passwordReset = 'passwordReset',
}

export interface ApiSuccess {
  type: OperationSuccess;
}
export interface UserContext<T = Prismeai.User | null> {
  user: T;
  loading: boolean;
  error?: ApiError;
  success?: ApiSuccess;
  signin: (email: string, password: string) => Promise<Prismeai.User | null>;
  signup: (
    email: string,
    password: string,
    firstName: string,
    lastName: string
  ) => Promise<Prismeai.User | null>;
  signout: (onServer?: boolean) => void;
  sendPasswordResetMail: (email: string, language: string) => Promise<any>;
  passwordReset: (token: string, password: string) => Promise<any>;
}

export const userContext = createContext<UserContext>({
  user: null,
  loading: false,
  signin: async () => null,
  signup: async () => null,
  signout() {},
  sendPasswordResetMail: async () => null,
  passwordReset: async () => null,
});

export function useUser(throwIfNotExist?: boolean): UserContext;
export function useUser(throwIfNotExist?: true): UserContext<Prismeai.User>;
export function useUser(throwIfNotExist?: boolean) {
  const context = useContext(userContext);

  if (throwIfNotExist && !context.user) {
    throw new Error();
  }

  return context;
}

export default userContext;
