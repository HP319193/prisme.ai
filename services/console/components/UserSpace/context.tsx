import { createContext } from 'react';
import { useContext } from '../../utils/useContext';

export interface UserSpaceConfig {
  /**
   * Main logo alternative
   */
  mainLogo?: {
    /**
     * Logo URL
     */
    url: string;
    /**
     * Attributes to apply on <img /> tag like alt or title
     */
    attrs?: object;
  };
  /**
   * Main URL on main logo link
   * @example: /product/ai-knowledge-chat/assistant?id=6474c0db33959b6283770367
   */
  mainUrl?: string;
  /**
   * The studio can act as a kiosk to keep the user in a URL template
   * @example: /product/ai-knowledge-chat/
   */
  kiosk?: string;
  /**
   * Define if Whats new popover is visible.
   * @default true
   */
  displayWhatsNew?: boolean;
  /**
   * Define if Help popover is visible.
   * @default true
   */
  displayHelp?: boolean;
  /**
   * Define if products popover is visible.
   * @default true
   */
  displayProducts?: boolean;

  /**
   * style object that will be injected inside various sections, allowing to cutomize theme colors ...
   */
  style?: {
    root?: object;
  };

  /**
   * Disable Builder
   */
  disableBuilder?: true;
}

interface UserSpaceProviderContext extends UserSpaceConfig {}
export const userSpaceContext = createContext<
  UserSpaceProviderContext | undefined
>(undefined);
export const useUserSpace = () =>
  useContext<UserSpaceProviderContext>(userSpaceContext);
export default userSpaceContext;
