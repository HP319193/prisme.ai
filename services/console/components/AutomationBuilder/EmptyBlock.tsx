import { FC, memo } from 'react';
import Image from 'next/image';
import { Handle, NodeProps, Position } from 'react-flow-renderer';
import { useAutomationBuilder } from './context';
import plus from '../../icons/plus.svg';

export const EmptyBlock: FC<NodeProps> = ({ data = {}, id }) => {
  const { addInstruction } = useAutomationBuilder();

  return (
    <>
      <Handle
        type="target"
        position={Position.Top}
        style={{ visibility: 'hidden' }}
      />
      <Handle
        type="source"
        position={Position.Bottom}
        style={{ visibility: 'hidden' }}
      />
      <div
        className="
          flex
          flex-col
          w-[260px]"
      >
        <div className="flex justify-center" style={{ minHeight: '1px' }}>
          {data.withButton && (
            <button
              className="flex items-center justify-center bg-graph-accent text-white rounded w-[1.625rem] h-[1.625rem] !rounded-[0.3rem]"
              onClick={() => {
                addInstruction(data.parent, data.index);
              }}
            >
              <Image src={plus.src} width={17} height={17} alt="" />
            </button>
          )}
        </div>
      </div>
    </>
  );
};

export default memo(EmptyBlock);
