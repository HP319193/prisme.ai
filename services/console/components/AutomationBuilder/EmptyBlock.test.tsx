import EmptyBlock from './EmptyBlock';
import renderer, { act } from 'react-test-renderer';
import { useAutomationBuilder } from './context';

jest.mock('react-flow-renderer', () => {
  return {
    Handle: () => <div>Handle</div>,
    Position: {
      Bottom: 'bottom',
      Top: 'top',
    },
  };
});

jest.mock('./context', () => {
  const mock = {
    addInstruction: jest.fn(),
  };
  return {
    useAutomationBuilder: () => mock,
  };
});

it('should render', () => {
  const root = renderer.create(
    <EmptyBlock
      id="a"
      data={{}}
      type="instruction"
      selected={false}
      isConnectable={false}
    />
  );
  expect(root.toJSON()).toMatchSnapshot();
});

it('should add instruction', () => {
  const parent: any = [];
  const root = renderer.create(
    <EmptyBlock
      id="a"
      data={{
        parent,
        index: 0,
        withButton: true,
      }}
      type="instruction"
      selected={false}
      isConnectable={false}
    />
  );

  act(() => {
    root.root.findByType('button').props.onClick();
  });

  expect(useAutomationBuilder().addInstruction).toHaveBeenCalledWith(parent, 0);
});
