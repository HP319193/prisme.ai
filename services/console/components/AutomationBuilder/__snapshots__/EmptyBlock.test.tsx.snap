// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render 1`] = `
Array [
  <div>
    Handle
  </div>,
  <div>
    Handle
  </div>,
  <div
    className=" flex flex-column w-10rem"
  >
    <div
      className="flex justify-content-center"
      style={
        Object {
          "minHeight": "1px",
        }
      }
    />
  </div>,
]
`;
